src/main/java/org/apache/sling/distribution/agent/impl/PrivilegeDistributionRequestAuthorizationStrategy.java

```diff
@@ -105,5 +113,14 @@ private String getClosestParent(Session session, String path) throws RepositoryE
         return null;
     }
 
+    private static Privilege[] computePrivileges(AccessControlManager acMgr, String[] jcrPrivileges, String jcrPrivilege)
```

Comment:
I think the primary reason they are recomputed today is because `String`=>`Privilege` needs a JCR session (`AccessControlManager`, which is derieved from `Session`) which isn't available in `PrivilegeDistributionRequestAuthorizationStrategy()` (or `PrivilegeDistributionRequestAuthorizationStrategyFactory.activate`).

IMHO using a service session for doing that might be an overkill.

Reply:
Ah, good point. We could keep the string representation in a Set.